lib/index.js

```diff
@@ -355,52 +354,48 @@ Server.prototype.reload = function(mixed) {
 }
 
 Server.prototype.stop = function(cb) {
-  if (this.app.started) {
-    _.each(this.app.interfaces, function(intf) {
-      if (
-        intf !== null &&
-        typeof intf === 'object' &&
-        typeof intf.stop === 'function'
-      ) {
-        intf.stop()
-      }
-    })
-
-    debug('Closing server...')
+  return new Promise((resolve, reject) => {
+    if (!this.app.started) {
+      resolve(this)
+    } else {
+      _.each(this.app.interfaces, function(intf) {
+        if (
+          intf !== null &&
+          typeof intf === 'object' &&
+          typeof intf.stop === 'function'
+        ) {
+          intf.stop()
+        }
+      })
 
-    const that = this
-    this.app.server.close(function() {
-      debug('Server closed')
-      if (that.app.redirectServer) {
-        that.app.redirectServer.close(() => {
-          debug('Redirect server closed')
-          delete that.app.redirectServer
+      debug('Closing server...')
+
+      const that = this
+      this.app.server.close(function() {
+        debug('Server closed')
+        if (that.app.redirectServer) {
+          that.app.redirectServer.close(() => {
+            debug('Redirect server closed')
+            delete that.app.redirectServer
+            cb && cb()
```

Comment:
Do we need the callback now that we return Promise?

Reply:
I was worried about backwards compatibility. Not sure if anything external is using it.